after/syntax/javascript.vim

```diff
@@ -173,11 +173,13 @@ syn region styledDefinition contained transparent fold extend
       \          cssHacks,
       \          cssCustom.*,
       \          jsComment,jsTemplateExpression,
-      \          typescriptInterpolation,
+      \          typescriptInterpolation,typescriptTemplateSubstitution,
       \          styledAmpersand,styledNestedRegion
 syn region styledDefinitionArgument contained transparent start=+(+ end=+)+
       \ contains=styledDefinition
 
+syn cluster typescriptValue add=styledPrefix,jsFuncCall,styledTypescriptPrefix
+
```

Comment:
Does any of these additions change the behavior after the above regex-match change? Because if not, we might get by with just changing the regex to include your inline-definition of the interface.

Reply:
This is unrelated to the regex. This actually solves problem with variable assignment:
```typescript
const myDiv = styled.div`

`;
```
Here everything after `=` is `@typescriptValue` which didn't contain styled stuff, so it didn't match at all.